components/ResultListPage/PlayAgainButton/index.js

```diff
@@ -0,0 +1,50 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import { useState } from "react";
+
+export default function PlayAgainButton({
+  currentRound,
+  setCurrentRound,
+  setRoundPoints,
+}) {
+  const router = useRouter();
+
+  const handlePlayAgainClick = () => {
+    if (currentRound < 3) {
+      setCurrentRound((prevCurrentRound) => prevCurrentRound + 1);
+      setRoundPoints({});
+      router.push("/");
+    }
+  };
```

Comment:
Do you need that function here? Since you need currentRound, setCurrentRound, setRoundPoints as probs from higher components, it might make more sense to uplift the handlePlayAgainClick function and only give it as the single prop as onPlayAgain back to this component